File: csharp/DeveloperTestingSamples/Samples/Chapter16/MethodDuplication/DateUtils.cs
using System;

namespace Samples.Chapter16.MethodDuplication
{
    // Emulating every legacy system's own implementation of date utilities.
    // This code is actually not in the book, only its Java counterpart.
    public class DateUtils
    {
        public static long DiffTime(DateTime t1, DateTime t2)
        {
            if (t1.Date != t2.Date)
            {
                throw new ArgumentException("Incomparable dates");
            }
            return (t2.Hour - t1.Hour) * 60;
        }

        public static long DiffTime_Revised(DateTime t1, DateTime t2)
        {
            if (t1.Date != t2.Date)
            {
                throw new ArgumentException("Incomparable dates");
            }
            return (t2.Hour * 60 + t2.Minute) - (t1.Hour * 60 + t1.Minute);
        }
    }
}

File: java/src/main/java/chapter16/methodduplication/DateUtils.java
package chapter16.methodduplication;

import java.util.Date;

/**
 * Emulating every legacy system's own implementation of date utilities.
 * Even using deprecated methods :)
 */
public class DateUtils {
    public static long diffTime(Date t1, Date t2) {
        if (t1.getDate() != t2.getDate()) {
            throw new IllegalArgumentException(
                    "Dates must be equal for comparison to work");
        }
        return (t2.getHours() - t1.getHours()) * 60;
    }

    public static long diffTime_revised(Date t1, Date t2) {
        if (t1.getDate() != t2.getDate()) {
            throw new IllegalArgumentException(
                    "Dates must be equal for comparison to work");
        }
        return (t2.getHours() * 60 + t2.getMinutes())
                - (t1.getHours() * 60 + t1.getMinutes());
    }
}
